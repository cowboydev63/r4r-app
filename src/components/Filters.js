import React from 'react';
import PropTypes from 'prop-types';
import FilterBox from './FilterBox';
import { Theme } from '../theme';


class Filters extends React.PureComponent {
    static propTypes = {
        facets: PropTypes.object.isRequired,
        onChange: PropTypes.func.isRequired,
    }

    static defaultProps = {
        facets: {},
        onChange: () => {},
    }

    render(){
        const toolTypesTypeFilters = (this.props.facets['toolTypes'] && this.props.facets['toolTypes'].items) || [];
        const isToolTypeSelected = Object.entries(toolTypesTypeFilters).some(([key, obj]) => obj.selected);
        return (
            <Theme element="section" className="results__facets" aria-label="Search Filters">
                <FilterBox 
                    className="tool-types"
                    facet={ this.props.facets['toolTypes'] }
                    onChange={ this.props.onChange('toolTypes') }
                    isCloseable={ true }
                />
                <FilterBox
                    className="tool-subtypes"
                    facet={ this.props.facets['toolSubtypes'] }
                    onChange={ this.props.onChange('toolSubtypes') }
                    isVisible={ isToolTypeSelected }
                />                        
                <FilterBox 
                    facet={ this.props.facets['researchAreas'] }
                    onChange={ this.props.onChange('researchAreas') }
                />
                <FilterBox 
                    facet={ this.props.facets['researchTypes'] }
                    onChange={ this.props.onChange('researchTypes') }
                />
            </Theme>
        )
    }
}

export default Filters;