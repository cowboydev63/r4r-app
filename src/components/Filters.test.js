import React from 'react';
import { shallow } from 'enzyme';
import Filters from './Filters';

describe('Filters Component', () => {
  it('renders without crashing', () => {
    shallow(<Filters />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Filters/>);
    expect(wrapper).toMatchSnapshot();
  })
})