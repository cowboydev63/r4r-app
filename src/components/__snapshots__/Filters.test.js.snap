// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Component renders correctly 1`] = `
<Theme
  aria-label="Search Filters"
  className="results__facets"
  element="section"
>
  <FilterBox
    className="tool-types"
    isCloseable={true}
    isVisible={true}
    onChange={[Function]}
  />
  <FilterBox
    className="tool-subtypes"
    isCloseable={false}
    isVisible={false}
    onChange={[Function]}
  />
  <FilterBox
    isCloseable={false}
    isVisible={true}
    onChange={[Function]}
  />
  <FilterBox
    isCloseable={false}
    isVisible={true}
    onChange={[Function]}
  />
</Theme>
`;
